#! /usr/bin/env python

PACKAGE = "base_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp", double_t, 0, "Kp - Proportional", 10.0, 0.0, 50.0)

gen.add("Kd", double_t, 0, "Kd - Differential", 12.0, 0.0, 50.0)

gen.add("Ki", double_t, 0, "Ki - Integral", 0.0, 0.0, 10.0)

gen.add("Ko", double_t, 0, "Ko - Offset", 50.0, 0.0, 100.0)

gen.add("accel_limit", double_t, 0, "Acceleration limit", 1.0, 0.0, 10.0)

gen.add("odom_linear_scale_correction", double_t, 0, "Linear Scale Correction", 1.0, 0.0, 2.0)

gen.add("odom_angular_scale_correction", double_t, 0, "Angular Scale Correction", 1.0, 0.0, 2.0)

exit(gen.generate(PACKAGE, "saturnbot_controller", "SaturnbotController"))
